### ISSUE TITLE
String duplicate in  snippets.json

### ISSUE THREAD

**Issue by VlasovVitaly** [CleverRaven/Cataclysm-DDA#11693]
I found some almost equal strings in snippets.json
It's in GLAMOPOLITAN comercial snippet.

There is three variant of strings:

This string placed twice in file, on 626 and 646 lines
```
"GLAMOPOLITAN!  We've got ALL the latest tips!  Whether you want to know what the elite are eating, wearing or discussing, Glamopolitan is YOUR magazine!  So pick up a copy today and \"Sizzle Like A Star!\""
```

One more string(different), line 666:
```
"GLAMOPOLITAN!  We've got ALL the latest tips!  Whether you want to know what the elite are eating, wearing or discussing, Glamopolitan is YOUR magazine!  So pick up a copy today and \"Sizzle Like A Star\"!"
```

One more string(different), line 695:
```
"GLAMOPOLITAN!  We've got ALL the latest tips!  Whether you want to know what the elite are eating, wearing or discussing, Glamopolitan is YOUR magazine!  So pick up a copy today and \"Sizzle Like A Star\"!"
```

Difference in this strings in spaces and punctuation. I don't know which is right one.

May be same thing with POPULAR MECHANICS snippet (lines 627, 647, 667, 696).
Other Snippets?

**Comment by BevapDin** [comment]
If an item uses those snippets as its description, the games stores the hash of the snippet in the item instance. This makes sure the same item always has the same description when you look at it. If the snippet changes, the hash changes as well and the game will not be able to find the original snippet text for an existing item. New items will work fine, they will use the new snippets and their hashes.

Existing items (from before that change) will show an empty description, but that could be changed to show the original (non-random) item description instead.

I'm fairly certain (not tested) the game will *not* crash or otherwise behave badly if it can't find the snippet of a hash. It will definitely not affect the game mechanics (behavior of the items / interaction with them), it's just the description that might change.

That said, I'm all for consistency, the notes above is just so people are not surprised.

tl;dr: Make it consistent, but existing items won't show it.

**Comment by VlasovVitaly** [comment]
Ohh... I see. There is small problem here. All old strings goes in to translation template. Should i (or someone) somehow adopt json strings extractor to this behaviour?

**Comment by BevapDin** [comment]
That shouldn't be a problem. Old items won't find the hash anyway (no translation of the string needed anymore) and new item would use the new strings which are properly translated.

However, one could use this trick, I just came up with:

Each snippet is defined in a snippet category, new items request a random snippet from a specific category. One could move the "old" (wrong) snippet texts into another category (say, "old_snippets_dont_use"). They would still be loaded, the string extractor would still extract them to be translated etc., but no new item would use them (because new items requests strings from that category). Existing items could still request them and they would show up just fine in the item description because requesting the snippet that belongs to a hash does not care about the category at all.

tl;dr: Move the wrong snippet into a category "old_snippets_dont_use" instead of just deleting them and there won't be any of the problems I mentioned in my previous comment.

----

Or: leave it as it is and blame the advertisement company that puts out those fliers (-:

**Comment by KA101** [comment]
One thing: those ads show up in different newspaper pages, so a certain amount of duplication is necessary.  If you want them made more distinct, sure, but they exist for a semi-reasonable (for marketing values of "reasonable") reason.

**Comment by kevingranade** [comment]
Just a thought, 'rare misprint'. ;)
